subsys/bluetooth/audio/has.c

```diff
@@ -539,6 +612,77 @@ static uint8_t handle_read_preset_req(struct bt_conn *conn, struct net_buf_simpl
 	return 0;
 }
 
+static int set_preset_name(uint8_t index, const char *name, size_t len)
+{
+	struct has_preset *preset = NULL;
+
+	BT_DBG("index %d name_len %zu", index, len);
+
+	if (len < BT_HAS_PRESET_NAME_MIN || len > BT_HAS_PRESET_NAME_MAX) {
+		return -EINVAL;
+	}
+
+	/* Abort if there is no preset in requested index range */
+	preset_foreach(index, BT_HAS_PRESET_INDEX_LAST, preset_found, &preset);
+
+	if (preset == NULL) {
+		return -ENOENT;
+	}
+
+	if (!(preset->properties & BT_HAS_PROP_WRITABLE)) {
+		return -EPERM;
+	}
+
+	IF_ENABLED(CONFIG_BT_HAS_PRESET_NAME_DYNAMIC, (
+		utf8_lcpy(preset->name, name, len + 1 /* null-terminator */);
+	));
```

Comment:
Hmm, I do wonder if a `memcpy` wouldn't be better here, since you already check the length and thus avoid any truncation, as `memcpy` is more efficient than `utf8_lcpy` as it does not need to do proper UTF8 truncation.

Reply:
I recall you encouraged me to use `utf8_lcpy` instead of `memcpy`, because it ensures UTF-8 formatted string... Is that correct? If yes, maybe it's better to use it here...

Reply:
@Thalley fixed now

Reply:
We can also update `utf8_lcpy` to no do truncation if string length of the source string is less than the size of the destination buffer (i.e. when no truncation takes place).